web/src/typeahead_helper.ts

```diff
@@ -1034,15 +1034,33 @@ export function query_matches_person(
     query: string,
     person: UserPillData | UserOrMentionPillData,
 ): boolean {
-    if (typeahead.query_matches_string_in_order(query, person.user.full_name, " ")) {
+    if (
+        person.type === "broadcast" &&
+        typeahead.query_matches_string_in_order(query, person.user.full_name, " ")
+    ) {
         return true;
     }
-    if (person.type === "user" && Boolean(person.user.delivery_email)) {
-        return typeahead.query_matches_string_in_order(
-            query,
-            people.get_visible_email(person.user),
-            " ",
+    if (person.type === "user") {
+        person.user.name_with_diacritics_removed ??= typeahead.remove_diacritics(
+            person.user.full_name.toLowerCase(),
         );
+        const query_with_diacritics_removed = typeahead.remove_diacritics(query.toLowerCase());
```

Comment:
You're still using `typeahead.remove_diacritics` on every iteration of the loop over users, on the query.